### PR TITLE
Refactor

### PR DESCRIPTION
**Pull request by coderhaoxin**

### CODE REVIEW

File: examples/example.js [hunk 1]
```diff
@@ -3,15 +3,17 @@
...
 
-app.use(error());
+app.use(error({
+  template: __dirname + '/error.html'
+}));
 
 app.use(function *(){
   if (this.path === '/404') return;
-  
+
```
Comment (RavenHursT) [inline examples/example.js]: remove extra line?

File: index.js [hunk 2]
```diff
@@ -23,12 +26,16 @@ module.exports = error;
...
-  var path = opts.template || __dirname + '/error.html';
-  var render = swig.compileFile(path);
+  const path = opts.template || __dirname + '/error.html';
 
   // env
-  var env = process.env.NODE_ENV || 'development';
+  const env = process.env.NODE_ENV || 'development';
+
+  var cache = opts.cache;
+  if (null == cache) cache = 'development' != env;
```
Comment (RavenHursT) [inline index.js]: Are we sure `cache` will always be `null` here?  Could just be:

```
cache = (opts.cache) ?
  opts.cache :
  ('development' !== env);
```

**Comment by RavenHursT** [comment]
This PR looks fine to allow consumers to define their own templating engine.  But I don't think it solves https://github.com/koajs/error/issues/19

`swig`, a no longer maintained package is still being included as a dependency.  I would suggest finding a templating engine that's still being supported/widely used, defaulting to that, and removing `swig` entirely.  People can still use this PR's changes to use `swig` if they really want to...

**Comment by coderhaoxin** [comment]
> a no longer maintained package is still being included as a dependency.

It's a depDev for the example, not a `dep` :)

> I would suggest finding a templating engine that's still being supported/widely used, defaulting to that

SGTM, any advice? `ejs`?

**Comment by coderhaoxin** [comment]
@RavenHursT open a new issue to discuss better default engine :)

**Comment by RavenHursT** [comment]
Sounds like a plan.

**Comment by RavenHursT** [comment]
done. https://github.com/koajs/error/issues/23

**Pull request merged**